include/core/Layer.hpp

```diff
@@ -0,0 +1,147 @@
+#ifndef __NGEN_LAYER__
+#define __NGEN_LAYER__
+
+#include "LayerData.hpp"
+#include "Simulation_Time.hpp"
+
+#ifdef NGEN_MPI_ACTIVE
+#include "HY_Features_MPI.hpp"
+#else
+#include "HY_Features.hpp"
+#endif
+
+#include "Pdm03.h"
+
+namespace ngen
+{
+
+    class Layer
+    {    
+        public:
+
+        #ifdef NGEN_MPI_ACTIVE
+            typedef hy_features::HY_Features_MPI feature_type;
+        #else
+            typedef hy_features::HY_Features feature_type;
```

Comment:
```suggestion
            using feature_type = hy_features::HY_Features_MPI;
        #else
            using feature_type = hy_features::HY_Features;
```
There's no good reason to keep using the mixed-order C-style typedefs in modern C++ code.

Reply:
old habits ...

Reply:
Nevermind fixed